pkg/services/ngalert/client/client.go

```diff
@@ -70,3 +73,48 @@ func TimeRequest(ctx context.Context, operation string, coll instrument.Collecto
 		coll, toStatusCode, doRequest)
 	return response, err
 }
+
+// TracedClient instruments a request with tracing. It implements Requester.
+type TracedClient struct {
+	client Requester
+	tracer tracing.Tracer
+}
+
+func NewTracedClient(client Requester, tracer tracing.Tracer) *TracedClient {
+	return &TracedClient{
+		client: client,
+		tracer: tracer,
+	}
+}
+
+// Do executes the request.
+func (c TracedClient) Do(r *http.Request) (*http.Response, error) {
+	url := r.URL.Path
+	method := r.Method
+	name := fmt.Sprintf("HTTP %s %s", method, url)
```

Comment:
Nit. To lower the cardinality might want to use a static name, similar to https://github.com/grafana/grafana/blob/main/pkg/infra/httpclient/httpclientprovider/tracing_middleware.go. Url and method are in the attributes as well

Reply:
Done! But with one caveat - I allowed the name to be customizable at the client level.
Cardinality is still much lower as you don't get a separate name for every single endpoint, but there are two names instead of one in this PR now. This allows you to more easily filter by subcomponent in the explore view. (view at all the Loki requests together, etc)